compendium/DeclarativeServices/test/TestUpdateConfiguration.cpp

```diff
@@ -302,216 +308,203 @@ TEST_F(tServiceComponent,
-    for (size_t i = 0; i < 5; i++) {
-      instanceSet.emplace(serviceObjects.GetService());
-    }
-
-    // Use DS runtime service to validate the component state
-    compConfigs =
-      t->GetComponentConfigs(testBundle, componentName, compDescDTO);
-    EXPECT_EQ(compConfigs.at(0).state, scr::dto::ComponentState::ACTIVE)
-      << "The state should be ACTIVE.";
-
-    EXPECT_TRUE(
-      std::none_of(instanceSet.begin(),
-                   instanceSet.end(),
-                   [](const std::shared_ptr<test::CAInterface>& service) {
-                     return service == nullptr;
-                   }));
-    EXPECT_EQ(instanceSet.size(), 5)
-      << "number of service instances returned must be equal to the number of "
-         "GetService calls";
+  compConfigs = GetComponentConfigs(testBundle, componentName, compDescDTO);
+  EXPECT_EQ(compConfigs.at(0).state, scr::dto::ComponentState::SATISFIED)
+    << "Component state should be SATISFIED";
+
+  instanceSet.clear();
+  // Call GetService to make the component active again. 
+  sRef = context.GetServiceReference<test::CAInterface>();
+  EXPECT_TRUE(static_cast<bool>(sRef));
+  serviceObjects = context.GetServiceObjects(sRef);
+  for (size_t i = 0; i < 3; i++) {
+    instanceSet.emplace(serviceObjects.GetService());
   }
 
-  // Confirm component instances were updated with the correct properties.
+  // Confirm component state is ACTIVE and all of the instances have the 
+  // updated property.
+  compConfigs = GetComponentConfigs(testBundle, componentName, compDescDTO);
+  EXPECT_EQ(compConfigs.size(), 1ul) << "One default config expected.";
+  EXPECT_EQ(compConfigs.at(0).state, scr::dto::ComponentState::ACTIVE)
+    << "Component instance state should be ACTIVE.";
+
   for (std::shared_ptr<test::CAInterface> service : instanceSet) {
     auto instanceProps = service->GetProperties();
     auto uniqueProp = instanceProps.find("uniqueProp");
-
     EXPECT_TRUE(uniqueProp != instanceProps.end())
       << "uniqueProp not found in constructed instance.";
-    EXPECT_EQ(uniqueProp->second, newInstanceId);
+    EXPECT_EQ(uniqueProp->second, updatedProp);
   }
-
-  instanceSet.clear();
-}
-
-/**
-  * Verify a service specified with scope as BUNDLE in component description
-  * is published with the correct scope and calls to GetService from the same
-  * bundle context return the same instance but calls from different contexts
-  * result in different instances of the service
-  */
-TEST_F(tServiceComponent, testUpdateConfig_BundleScope_Modified) //DS_CAI_FTC_26
-{
-  std::string componentName = "sample::ServiceComponentCA26";
-  std::string testBundleName = "TestBundleDSCA26";
-  scopeValidation(componentName,
-                  testBundleName,
-                  true,
-                  cppmicroservices::Constants::SCOPE_BUNDLE,
-                  2);
-}
-
-TEST_F(tServiceComponent,
-       testUpdateConfig_BundleScope_WithoutModified) //DS_CAI_FTC_27
```

Comment:
This test was re-factored heavily and I don't see these two test points (testUpdateConfig_BundleScope_Modified and testUpdateConfig_BundleScope_WithoutModified)  in this file.  are these tests still needed?

Reply:
The code being tested in DS is in BundleOrPrototypeComponentConfiguration.cpp in the construction of the component instance and the modification of a component instance. Bundle and Prototype both share all of this code so only one set of tests is needed. The BundleScope tests weren't testing any additional lines of code. The other problem with the BundleScope tests was that the developer created a second framework to test them. Not sure if there is another way to test a service with Bundle Scope but testing it by creating a second framework was problematic.